### PR TITLE
Add pause and resume to chronometer

### PR DESCRIPTION
**Pull request by Neyune**
We did not implemented the Pause method, because it's exactly the same as Stop

### CODE REVIEW

File: includes/system/spk_chronometer.hpp [hunk 1]
```diff
@@ -19,14 +21,28 @@ namespace spk
...
+		bool isRunning();
+
 		/**
 		 * @brief Get the duration since the start of the chronometer
 		 * @return The duration in milliseconds
 		 * @note Return 0 if the chronometer is not started.
 		 * @note If stop has been called, the duration is not updated.
 		 */
-		const long long& duration() const;
+		const long long duration() const;
```
Comment (NotCampeur) [inline includes/system/spk_chronometer.hpp]: I can understand the deletion of the reference for safety purpose, but by doing so the const keyword for the return need to be deleted too.

File: src/system/spk_chronometer.cpp [hunk 2]
```diff
@@ -14,24 +16,44 @@ namespace spk
...
+			throw std::runtime_error("Can't start an already started chronometer");
+
 		_start = spk::TimeMetrics::instance()->time();
 		_duration = 0;
+		_totalDuration = 0;
+		_isRunning = true;
+	}
+
+	void Chronometer::resume()
+	{
```
Comment (NotCampeur) [inline src/system/spk_chronometer.cpp:29]: You might want to add a check for _totalDuration here.
The user can resume a Chronometer that hasn't started.

**Review by NotCampeur — CHANGES_REQUESTED** [review]
Good implementation.
Please address the comments.

File: src/system/spk_chronometer.cpp (hunk 2 above)
Comment (Neyune) [reply src/system/spk_chronometer.cpp:29]: We added the isRunning so it's fine

File: includes/system/spk_chronometer.hpp (hunk 1 above)
Comment (Neyune) [reply includes/system/spk_chronometer.hpp]: const has been removed

**Review by NotCampeur — APPROVED** [review]
Thanks for the fixes.

**Pull request merged**